deltachat-ios/AppDelegate.swift

```diff
@@ -461,7 +459,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
             self.dcAccounts.startIo()
             self.dcAccounts.maybeNetwork()
 
-            _ = fetchSemaphore.wait(timeout: .now() + 10)
+            // create a new semaphore to make sure the received DC_CONNECTIVITY_CONNECTED really belongs to maybeNetwork() from above
+            // (maybeNetwork() sets connectivity to DC_CONNECTIVITY_CONNECTING, when fetch is done, we're back at DC_CONNECTIVITY_CONNECTED)
+            self.dcAccounts.fetchSemaphore = DispatchSemaphore(value: 0)
```

Comment:
from what I understood, there are cases where perform fetch can be called twice right after each other. it would lead to one semaphore is thrown away immediately. I'm not sure about the implications in depth. It might be ok. However we could also add a null check here.

Reply:
i think, this is just okay, would result in one thread waiting for the max. time and not getting a signal (the situations where there are really two concurrent performFetch running are very rare)

i'd like to avoid an `fetchSemaphore != nil` check to avoid weird other happenings as never getting performFetch running - same if we would reuse semaphores and there are too many/few singal() raised.